docs/plugin-eval.md

```diff
@@ -0,0 +1,221 @@
+
+## Getting Started
+
+To create a new plugin, follow these steps:
+
+1. Familiarize yourself with the example plugins found in the `openai` folder, such as `openai/bird.py`, `openai/clock.py`, and `openai/text_analyzer.py`. These examples will help you understand the structure and requirements for creating a new plugin.
+
+2. Follow the plugin creation process detailed below.
+
+3. Submit a pull request with your new plugin and evaluation example(s).
+
+## Preparing Samples
+
+Evaluations with plugins enabled take an additional arguement - `plugins` - which is a list of plugins to enable. The evaluation framework will automatically instantiate the plugins from the registry for each sampling.  Please note, we currently require chat-style inputs for plugins.
+
+Before sending a request to the model, we will iterate through the `input` messages, invoking the relevant plugins.  One plugin instance will be used per sample, so plugins will retain state throughout that request.
+
+NOTE: By enabling plugins, the input list may be mutated.  Currently, we append the plugin's description to the end of the system message (Or insert a new system message if none was provided). This is done to ensure that the model has context to know how to use the plugin.
+
+```JSON
+{
+  "plugins": [
+    "openai.bird.1",
+    "openai.clock.1",
+  ],
+  "ideal": [
+    "Red-tailed hawk"
+  ],
+  "input": [
+    {
+      "role": "system",
+      "content": "Answer the following questions as concisely as possible."
+    },
+    {
+      "role": "user",
+      "content": "Can you list all of the birds in my birds list which are in the Accipitridae family?"
+    },
+    {
+      "role": "plugin",
+      "namespace": "MyBirds",
+      "endpoint": "listBirds",
+      "content": {}
+    }
+  ]
+}
+```
+
+## Plugin Creation
+
+### Namespace and API Description
+
+- Use the `@namespace` decorator to define your plugin's namespace and provide a brief description. The AI model will use this namespace to encapsulate your plugin. The description will help the model understand the goals, purposes, and limitations of your plugin.
+
+- For each API method in your plugin, use the `@api_description` decorator to define the API's name, description, and arguments (if any). The AI model will use this information to understand the purpose of each API method and the inputs it expects. The description will also provide context for the function.
+
+### Input and Output
+
+- Ensure your plugin methods accept inputs as defined in the `api_args` parameter of the `@api_description` decorator.
+
+- The output of your plugin methods should be in a format that can be easily consumed by the AI model.
```

Comment:
Could you give a few examples of what is an easily consumable format? What are the types of outputs we support?